Update dependencies for security reasons

PierreRambaud opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Update dependencies for security reasons.<br><br>**prestashop/ps_facetedsearch**: v3.0.6 -> 3.5.0<br>**prestashop/ps_linklist**: 3.0.3 -> v3.1.0<br>**prestashop/ps_socialfollow**: v2.0.0 -> v2.1.0
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Check installed modules.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/18651)
<!-- Reviewable:end -->

eternoendless reviewed (changes requested):
PR description must state exactly which dependencies have been updated

eternoendless commented on tests-legacy/Integration/PrestaShopBundle/Controller/Api/StockManagementControllerTest.php:
```diff
@@ -212,7 +212,7 @@ private function assertResponseHasTotalPages($parameters, $expectedTotalPages)
         $headers = $response->headers;
         $this->assertTrue($headers->has('Total-Pages'), 'The response headers should contain the total pages.');
-        $this->assertSame(
+        $this->assertEquals(
             $expectedTotalPages,
```
Maybe you should change `$expectedTotalPages` instead

PierreRambaud replied:
The problem is this value came from Symfony, we shouldn't test it.
Since we are working with number, we don't care if Symfony returns `3` or `'3'`
We already have this debate for the 1.7.7.x  https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/tests-legacy/Integration/PrestaShopBundle/Controller/Api/StockManagementControllerTest.php#L215

Commit:
Update dependencies

    prestashop/ps_facetedsearch: v3.0.6 -> 3.5.0
    prestashop/ps_linklist: 3.0.3 -> v3.1.0
    prestashop/ps_socialfollow: v2.0.0 -> v2.1.0

eternoendless commented:
Thank you @PierreRambaud
